src/lib.rs

```diff
@@ -178,32 +184,60 @@ impl LightningNode {
 
         // Step 11: Optional: Initialize the NetGraphMsgHandler
         let _graph = persister.read_graph();
+        let genesis = genesis_block(config.network).header.block_hash();
+        let graph = Arc::new(NetworkGraph::new(genesis, Arc::clone(&logger)));
+        let rapid_gossip = Arc::new(RapidGossipSync::new(Arc::clone(&graph)));
 
         // Step 12. Initialize the PeerManager
-        let _peer_manager = init_peer_manager(
+        let peer_manager = Arc::new(init_peer_manager(
             Arc::clone(&channel_manager),
             &keys_manager,
             Arc::clone(&logger),
-        );
+        )?);
 
         // Step 13. Initialize Networking
 
         // Step 14. Keep LDK Up-to-date with Chain Info
 
         // Step 15. Initialize an EventHandler
-        let _event_handler = LipaEventHandler {};
+        let event_handler = Arc::new(LipaEventHandler {});
 
         // Step 16. Initialize the ProbabilisticScorer
         let _scorer = persister.read_scorer();
+        let scorer = Arc::new(Mutex::new(ProbabilisticScorer::new(
+            ProbabilisticScoringParameters::default(),
+            Arc::clone(&graph),
+            Arc::clone(&logger),
+        )));
 
         // Step 17. Initialize the InvoicePayer
 
         // Step 18. Initialize the Persister
         // Persister trait already implemented and instantiated ("persister")
 
         // Step 19. Start Background Processing
+        // This assumes that the background processor will never fail.
```

Comment:
Could you elaborate on what you mean by *this*?

Do you mean the fact that we just `start()` and then never join and restart on potential errors?

Reply:
I elaborated in the comment.